Agents/Ifc2TilesetAgent/README.md

```diff
@@ -0,0 +1,160 @@
+    - Once the container has been built and the `launch.json` is available, click the green arrow at the top left corner to run the debugger. Debugging capabilities should now be available.
+
+**PRODUCTION ENVIRONMENT**
+- Deploy the agent and its dependencies by running the following code in the command prompt at the `<root>` directory:
+```
+docker-compose up -d 
+```
+
+## 2. Running the agent
+### 2.1 Precursor
+Place only one IFC file in `<root>\data\ifc\`. This directory is directly linked to the relevant directory in the Docker container. The agent is only able to convert ONE IFC model at a time.
+
+Please modify the following properties in `config/properties.yaml`:
+
+- `query_endpoint`^ : SPARQL endpoint for Query operations
+- `update_endpoint`^ : SPARQL endpoint for UPDATE operations
+
+^*Endpoints are required to query for metadata in tileset and interactions during visualisation*
+
+### 2.2 API
+Instructions for the agent and its various API routes can be found at the API root `http://localhost:5105/`. Users can visit this route in any browser to verify if the agent is running.
+
+A brief overview is as follows:
+- **POST** request to convert an IFC model to 3D Tileset, and output tileset.json files. Accepted parameters:
+    1. `assetUrl`  
+    - Sets the file path to directory or url holding the glb assets in the tilesets generated.
+    - Valid formats include `"."`, `"./file/path"`, `"../../file/path"`, and `"http://www.example.com"`. Please do not add an ending `/`, which will be generated in the code itself.
+```
+/api
+```
+
+### 2.3 POST Request
+Run the agent by sending a POST request with the required JSON Object to the necessary endpoint. A sample request in `curl` syntax is as follows:
+```
+curl -X POST localhost:5105/api -H 'Content-Type: application/json' -d '{\"assetUrl\":\"./glb\"}'  
+```
+
+If the agent ran successfully, a JSON Object would be returned as follows:
+```
+{"Result":["IFC model has successfully been converted. Please visit the 'data' directory for the outputs"]}
+```
+
+## 3. For Developers
+The agent have been packaged into the following submodules:
+- `agent/app.py` is the agent's entry point that accepts and processes all HTTP requests
+- `agent/config` contain functions to retrieve the properties in `config/properties.yaml`
+- `agent/exceptions` contain the exceptions encountered in this agent
+- `agent/kgutils` contain miscellaneous functions to access and query the Knowledge graph
+- `agent/utils` contain miscellaneous functions for searching, validating inputs, and system operations
+- `agent/ifc2gltf` contain functions to query the metadata, and process the IFC input into their individual geometry files for each asset
+- `agent/ifc2tileset` contain functions to generate the tileset and write them to json
+
+As Git does not allow empty directories, `.gitignore` files have been added to the subdirectories  of `<root>\data\`. This is important to set up the file structure for the code to run. 
```

Comment:
FYI: It is customary to put empty `.gitkeep` files into folders that would otherwise be empty for this purpose.